### ISSUE TITLE
[Bug]: VAMB on AMD GPU (Pawsey)

### ISSUE THREAD

**Issue by mpdoane2** [linsalrob/atavide_lite#44]
What happened?

The current atavide_lite_vamb.yaml installs VAMB through pip without specifying the required PyTorch accelerator backend.

On Pawsey Setonix, this causes pip to install the default NVIDIA CUDA build of PyTorch 2.6.0:

nvidia-cublas-cu12       12.4.5.8
nvidia-cuda-cupti-cu12   12.4.127
nvidia-cuda-nvrtc-cu12   12.4.127
nvidia-cuda-runtime-cu12 12.4.127
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.1.3
nvidia-curand-cu12       10.3.5.147
nvidia-cusolver-cu12     11.6.1.9
nvidia-cusparse-cu12     12.3.1.170
nvidia-cusparselt-cu12   0.6.2
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.4.127
torch                    2.6.0

Setonix uses AMD Instinct MI250X GPUs with ROCm rather than NVIDIA GPUs with CUDA. Consequently, importing the default PyTorch installation hangs and VAMB cannot run.

Replacing the NVIDIA PyTorch build with the ROCm 6.2.4 build resolves the issue.

How can we reproduce it?
Create the VAMB environment from the existing YAML:
VAMB_ENV="/scratch/$PAWSEY_PROJECT/$USER/software/miniconda3/atavide_lite_vamb"

mamba env create \
  --yes \
  --prefix "$VAMB_ENV" \
  --file ../atavide_lite_vamb.yaml
Inspect the PyTorch installation:
"$VAMB_ENV/bin/python" -m pip show torch

"$VAMB_ENV/bin/python" -m pip list |
  grep -Ei 'torch|cuda|nvidia|rocm|hip'

This shows that PyTorch 2.6.0 and its NVIDIA CUDA 12 dependencies were installed.

Test the PyTorch import:
timeout 120s "$VAMB_ENV/bin/python" -c '
print("Importing PyTorch...", flush=True)
import torch
print("PyTorch:", torch.__version__, flush=True)
'

echo "exit status: $?"

The import does not complete and is terminated by timeout.

For comparison, NumPy imports successfully:
timeout 60s "$VAMB_ENV/bin/python" -c '
import numpy
print("NumPy:", numpy.__version__, flush=True)
'

echo "exit status: $?"
Error output
Importing PyTorch...
exit status: 124

Exit status 124 indicates that the PyTorch import exceeded the specified timeout.

Environment
atavide_lite commit or version: [add commit or version]
Pipeline profile: pawsey_shortread
Cluster and scheduler: Pawsey Setonix, Slurm
GPU architecture: AMD Instinct MI250X
Accelerator platform: AMD ROCm
Operating system: Setonix software environment
Conda environment: atavide_lite_vamb
Python version: 3.11
VAMB version: [add VAMB version]
PyTorch initially installed: 2.6.0 with NVIDIA CUDA 12 dependencies
Input type: paired short reads
Workaround

Removing the NVIDIA build and installing the ROCm 6.2.4 build of PyTorch resolves the import problem:

VAMB_ENV="/scratch/$PAWSEY_PROJECT/$USER/software/miniconda3/atavide_lite_vamb"

"$VAMB_ENV/bin/python" -m pip uninstall -y \
  torch triton \
  nvidia-cublas-cu12 \
  nvidia-cuda-cupti-cu12 \
  nvidia-cuda-nvrtc-cu12 \
  nvidia-cuda-runtime-cu12 \
  nvidia-cudnn-cu12 \
  nvidia-cufft-cu12 \
  nvidia-curand-cu12 \
  nvidia-cusolver-cu12 \
  nvidia-cusparse-cu12 \
  nvidia-cusparselt-cu12 \
  nvidia-nccl-cu12 \
  nvidia-nvjitlink-cu12 \
  nvidia-nvtx-cu12

"$VAMB_ENV/bin/python" -m pip install \
  --index-url https://download.pytorch.org/whl/rocm6.2.4 \
  "torch==2.6.0"

The corrected installation reports:

2.6.0+rocm6.2.4

Both PyTorch and VAMB then import successfully.

Suggested solution

A Pawsey-specific environment or installation step could explicitly install the ROCm-compatible PyTorch build before installing VAMB. This would prevent pip install vamb from selecting the default NVIDIA CUDA build on Setonix.

**Comment by linsalrob** [comment]
Yes, use the pawsey version of `vamb_install.slurm`. Pawsey uses a different type of GPU (called AMD) than most other systems (which use NVIDIA) and hence we need to install a slightly different version of the software. The pawsey vamb install works great.
